### PR TITLE
 #99: Only use `CMAKE_POSITION_INDEPENDENT_CODE` when building bindings

### LINKED ISSUE #99: Only use `CMAKE_POSITION_INDEPENDENT_CODE` when building bindings
`CMAKE_POSITION_INDEPENDENT_CODE=ON`  is only required when we build with nanobind.

Enable it conditionally so that `vt` (which will never use the bindings) doesn't have to worry about it.

### PR DESCRIPTION
**Pull request by cwschilly**
Fixes #99

### CODE REVIEW

**Comment by pierrepebay** [comment]
Looks good. Out of curiosity, what is the error when building the bindings without this option turned on?

**Comment by cwschilly** [comment]
> Looks good. Out of curiosity, what is the error when building the bindings without this option turned on?

@pierrepebay 

```
[ 98%] Linking CXX shared module vttv.cpython-39-x86_64-linux-gnu.so
/usr/bin/ld: ../../lib/fmt/libfmt-vtd.a(format.cc.o): warning: relocation against `_ZTVN3fmt2vt12system_errorE' in read-only section `.text._ZN3fmt2vt12system_errorC2IJEEEiNS0_17basic_string_viewIcEEDpRKT_[_ZN3fmt2vt12system_errorC5IJEEEiNS0_17basic_string_viewIcEEDpRKT_]'
/usr/bin/ld: ../../src/libvt-tv.a(render.cc.o): relocation R_X86_64_PC32 against symbol `_ZN2vt2tv14compareObjectsERKSt4pairINS0_10ObjectWorkEmES5_' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [bindings/python/CMakeFiles/vttv.dir/build.make:142: bindings/python/vttv.cpython-39-x86_64-linux-gnu.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:1784: bindings/python/CMakeFiles/vttv.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,17 +24,19 @@ set(
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-# add -fPIC to all targets
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 17 CACHE STRING "The C++ standard to use")
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "The C++ standard to use")
```
Comment (nlslatt) [inline CMakeLists.txt:28]: What's the reason for the whitespace change?

Comment (cwschilly) [reply CMakeLists.txt]: My mistake; it's fixed now

**Review by nlslatt — APPROVED** [review]
I'm not sure if there's another way to fix the link error, but this works at least for now.

**Pull request merged**